internal/mode/static/state/graph/policies.go

```diff
@@ -72,18 +72,18 @@ func (g *Graph) attachPolicies(ctlrName string) {
 	}
 
 	for _, policy := range g.NGFPolicies {
-		ref := policy.TargetRef
-
-		switch ref.Kind {
-		case kinds.Gateway:
-			attachPolicyToGateway(policy, g.Gateway, g.IgnoredGateways, ctlrName)
-		case kinds.HTTPRoute, kinds.GRPCRoute:
-			route, exists := g.Routes[routeKeyForKind(ref.Kind, ref.Nsname)]
-			if !exists {
-				continue
-			}
+		for _, ref := range policy.TargetRefs {
```

Comment:
will it work well if policy targets both Gateway and HTTPRoute at the same time?

Reply:
That will come down to the policy design itself I think. One of the reasons why we don't yet want to support multiple refs on the ClientSettingsPolicy. Needs to be something we could apply at both levels and still have it make sense.